src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -445,9 +445,21 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             }
 
             ImmutableArray<PendingBranch> pendingReturns = base.Scan(ref badRegion);
+            EnforceDoesNotReturn(methodMainNode.Syntax.GetLastToken().GetLocation());
```

Comment:
>methodMainNode.Syntax.GetLastToken().GetLocation() [](start = 33, length = 50)

Consider passing in the method syntax instead rather than calling `GetLocation()` eagerly.

Reply:
I don't think I could do that, because the check on end of method uses the last token of the syntax, instead of a `SyntaxNode`

Reply:
I don't follow. Couldn't the method be defined as:
```C#
void EnforceDoesNotReturn(SyntaxNode syntax)
{
    ...
    ReportDiagnostic(ErrorCode.WRN_ShouldNotReturn,
        syntax.GetLastToken().GetLocation());
}

---
In reply to: [369381283](https://github.com/dotnet/roslyn/pull/41098#discussion_r369381283) [](ancestors = 369381283)